ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2835,232 @@ public sealed class Bot : IAsyncDisposable {
+			}
+
+			return appIDs.ToHashSet();
+		}
+
+		private async Task<List<uint>> GetCompletedBadgeAppIDs(byte page) {
+			using IDocument? badgePage = await ArchiWebHandler.GetBadgePage(page).ConfigureAwait(false);
+
+			if (badgePage == null) {
+				ArchiLogger.LogGenericWarning(Strings.WarningCouldNotCheckBadges);
+
+				return new List<uint>(0);
+			}
+
+			return GetCompletedBadgeAppIDs(badgePage);
+		}
+
+		private List<uint> GetCompletedBadgeAppIDs(IDocument badgePage) {
+			List<IElement> craftButtons = badgePage.SelectNodes("//a[@class='badge_craft_button']");
+
+			if (craftButtons.Count == 0) {
+				// no craftable badges
+				return new List<uint>(0);
+			}
+
+			List<uint> result = new List<uint>(craftButtons.Count);
+
+			foreach (string? badgeUri in craftButtons.Select(htmlNode => htmlNode.GetAttribute("href"))) {
+				if (string.IsNullOrEmpty(badgeUri)) {
+					ArchiLogger.LogNullError(nameof(badgeUri));
+
+					continue;
+				}
+
+				// URIs to foil badges are the same as for normal badges except they end with "?border=1"
+				string? appIDText = badgeUri.Replace("?border=1", "").Split('/', StringSplitOptions.RemoveEmptyEntries).Last();
+
+				if (!uint.TryParse(appIDText, out uint appID) || (appID == 0)) {
+					ArchiLogger.LogNullError(nameof(appID));
+
+					continue;
+				}
+
+				result.Add(appID);
+			}
+
+			return result;
+		}
+
+		private async Task SendCompletedSets() {
+			ISet<uint> appIDs = await GetCompletedBadgeAppIDs().ConfigureAwait(false);
+
+			if (appIDs.Count == 0) {
+				// Could not extract appIds for craftable badges - No need to load the inventory at this point
+				return;
+			}
+
+			HashSet<Steam.Asset> inventory;
+
+			try {
+				inventory = await ArchiWebHandler.GetInventoryAsync(SteamID)
+					.Where(item => item.Tradable)
+					.Where(item => (item.Type == Steam.Asset.EType.TradingCard) || (item.Type == Steam.Asset.EType.FoilTradingCard))
+					.Where(item => appIDs.Contains(item.RealAppID))
+					.ToHashSetAsync()
+					.ConfigureAwait(false);
+			} catch (HttpRequestException e) {
+				ArchiLogger.LogGenericWarningException(e);
+
+				return;
+			} catch (Exception e) {
+				ArchiLogger.LogGenericException(e);
+
+				return;
+			}
+
+			if (inventory.Count == 0) {
+				ArchiLogger.LogGenericWarning(string.Format(Strings.ErrorIsEmpty), nameof(inventory));
+
+				return;
+			}
+
+			HashSet<Steam.Asset> result = new HashSet<Steam.Asset>();
+
+			foreach (uint appID in appIDs) {
+				HashSet<Steam.Asset> itemsForAppBadges = await GetItemsForFullBadge(inventory, appID).ConfigureAwait(false);
+
+				result.UnionWith(itemsForAppBadges);
+			}
+
+			if (result.Count > 0) {
+				await Actions.SendInventory(inventory: result).ConfigureAwait(false);
+			}
+		}
+
+		private async Task<HashSet<Steam.Asset>> GetItemsForFullBadge(IReadOnlyCollection<Steam.Asset> inventory, uint appID) {
+			byte requiredCards = await ArchiWebHandler.GetCardCountForGame(appID).ConfigureAwait(false);
```

Comment:
Consider cache values since a lot of badges will cause sending lots of requests to Steam every time.